library/blob.c

```diff
@@ -2028,99 +2026,128 @@ static int eblob_write_prepare_disk(struct eblob_backend *b, struct eblob_key *k
 			copy, copy_offset, old);
 
 err_out_exit:
+	return err;
+}
+
+
+/**
+ * eblob_write_prepare_disk() - allocates space for new record
+ * It allocates new bases, commits headers and manages overwrites/appends.
+ * NB! Function takes "backend" lock.
+ */
+static int eblob_write_prepare_disk(struct eblob_backend *b, struct eblob_key *key,
+		struct eblob_write_control *wc, uint64_t prepare_disk_size,
+		enum eblob_copy_flavour copy, uint64_t copy_offset, struct eblob_ram_control *old,
+		size_t defrag_generation)
+{
+	pthread_mutex_lock(&b->lock);
+	int err = eblob_write_prepare_disk_nolock(b, key, wc, prepare_disk_size,
+	                                          copy, copy_offset, old, defrag_generation);
 	pthread_mutex_unlock(&b->lock);
-	eblob_dump_wc(b, key, wc, "eblob_write_prepare_disk", err);
 	return err;
 }
 
+
+/**
+ * write_prepare_inplace() - prepare space in existing base. Non-blocking.
+ */
+static int write_prepare_inplace(struct eblob_backend *b, struct eblob_key *key,
+		uint64_t flags, struct eblob_write_control *wc) {
+	// We've got a key which will be overwritten, make sure it has valid flags.
+	// We overwrite flags to what user has provided dropping all existing on-disk flags since
+	// given key will be fully overwritten and we do not care about its old content anymore.
+	// The same logic is performed in @eblob_write_prepare_disk(), but it also allocates new space.
+
+	int err = 0;
+	uint64_t new_flags = 0;
+	new_flags = eblob_validate_ctl_flags(b, flags);
+	new_flags |= BLOB_DISK_CTL_UNCOMMITTED;
+
+	if (wc->flags == new_flags) {
+		goto out_stat;
```

Comment:
Why isn't it just:
```c
eblob_stat_inc(b->stat, EBLOB_GST_PREPARE_REUSED);
goto out_exit;
```

Reply:
we want to inc stat in another success cases

Reply:
You are right, thanks